src/test_memcachedstoreview.cpp

```diff
@@ -0,0 +1,232 @@
+ *
+ * Project Clearwater - IMS in the cloud.
+ * Copyright (C) 2015  Metaswitch Networks Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version, along with the "Special Exception" for use of
+ * the program along with SSL, set forth below. This program is distributed
+ * in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * The author can be reached by email at clearwater@metaswitch.com or by
+ * post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+ *
+ * Special Exception
+ * Metaswitch Networks Ltd  grants you permission to copy, modify,
+ * propagate, and distribute a work formed by combining OpenSSL with The
+ * Software, or a work derivative of such a combination, even if such
+ * copying, modification, propagation, or distribution would otherwise
+ * violate the terms of the GPL. You must comply with the GPL in all
+ * respects for all of the code used other than OpenSSL.
+ * "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+ * Project and licensed under the OpenSSL Licenses, or a work based on such
+ * software and licensed under the OpenSSL Licenses.
+ * "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+ * under which the OpenSSL Project distributes the OpenSSL toolkit software,
+ * as those licenses appear in the file LICENSE-OPENSSL.
+ */
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include "memcachedstoreview.h"
+
+using ::testing::AnyOf;
+using ::testing::Contains;
+
+// Function that expects a valid replica list.
+//
+// @param - The list to check.
+// @param - The expected length of the list.
+// @param - An AnyOf matcher containing the allowed values for each entry in
+//          the list.
+template<class M>
+void expect_replica_list(MemcachedStoreView::ReplicaList replicas,
+                         size_t count,
+                         M matcher)
+{
+  // Check the count.
+  EXPECT_EQ(replicas.size(), count);
+
+  // Check each entry is unique. Do this by sorting the list, unique-ing it and
+  // checking the length has not changed.
+  std::sort(replicas.begin(), replicas.end());
+  std::unique(replicas.begin(), replicas.end());
+  EXPECT_EQ(replicas.size(), count);
+
+  // Check each entry of the list is allowed.
+  for (size_t i = 0; i < count; ++i)
+  {
+    EXPECT_THAT(replicas[i], matcher);
+  }
+}
+
+#define EXPECT_REPLICA_LIST(LIST, COUNT, ...)                                  \
```

Comment:
Could do with a comment, in particular, the `__VA_ARGS__` are not the replica list, they're the list that the replicas can be chosen from.